src/main/java/com/sivalabs/bookstore/catalog/domain/ProductService.java

```diff
@@ -30,4 +40,10 @@ public PagedResult<ProductModel> getProducts(int pageNo) {
     public Optional<ProductModel> getProductByCode(String code) {
         return productRepository.findByCode(code).map(productMapper::toModel);
     }
+
+    public Optional<Product> deleteProduct(String code) {
+        Query query = new Query(Criteria.where("code").is(code));
```

Comment:
Can we simplify the implementation as follows:
1. Fetch Product using productRepository.findByCode(code)
2. If product not exists throw ProductNotFoundException
3. Otherwise set deleted = true and productRepository.save(product)

Also, could we please add a test?